config/config.go

```diff
@@ -0,0 +1,171 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/googleinterns/aso_sxs_viewer/proto"
+	"google.golang.org/protobuf/encoding/prototext"
+)
+
+type ViewerConfig struct {
+	*proto.ViewerConfig
+}
+
+type BrowserConfig struct {
+	Selector string
+	Position int
+	URL      string
+}
+
+// GetConfig generates .aso_sxs_viewer directory and config file, if it doesn't already exists.
+// It validates the pre-existing config if found.
+// You should avoid repeated calls to avoid validation overhead.
+func GetConfig() (ViewerConfig, error) {
+	if err := createDir(AsoSxSViewerDir); err != nil {
+		return ViewerConfig{nil}, fmt.Errorf("Error %s encountered while creating aso_sxs_viewer directory", err.Error())
+	}
+
+	return createOrValidateConfig(AsoSxSViewerConfig)
+}
+
+func createDir(dirPath string) error {
+	// check if directory exists
+	_, err := os.Stat(dirPath)
+
+	// create directory if does not exist
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(dirPath, 0755)
+		if err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
```

Comment:
rather than the else-if check, just do `return err` (if error is nil, it automatically returns nil)